ls1intum/Artemis
# Show the correct build logs for specific results

### Checklist

### Motivation and Context
This resolves #3849. The problem here is, that build logs are stores based on the latest build for each individual submission, but the build logs are only available on participation level. This results in older failed submissions showing the latest submission's build logs (which might not exist).

### Description
- Added a `resultId` as an optional parameter, as discussed in the issue.
- Added integration tests for result-specific build logs.
- Make all client and server tests pass with the changes.

### Steps for Testing
- Make sure the issue cannot be reproduced anymore, you can also be creative here and check what happens with multiple failed submissions. Just be aware that always the latest result of each submission counts.
- You still get the build logs in the online code editor when working on a programming exercise, and it fails (e.g. due to compilation errors)

### Review Progress





- Code Review
- Manual Tests

### Test Coverage
- Server almost 100%:
  ![image](https://user-images.githubusercontent.com/11130248/129783024-58621eff-f924-4937-b5cd-0c9905196d0b.png)
- Client service now at 100%
  ![image](https://user-images.githubusercontent.com/11130248/129767336-0dd39a4e-5cf9-4d24-bd4a-fa5f7ccbe2aa.png)

### Screenshots
![image](https://user-images.githubusercontent.com/11130248/129770138-998f7f54-8ae7-4b70-b1c7-0004aa3ab3d0.png)
